docs/dev/table/catalog.md

```diff
@@ -0,0 +1,367 @@
+
+## Dependencies
+
+To use `HiveCatalog`, users need to include the following dependency jars.
+
+For Hive 2.3.4, users need:
+
+```
+// Hive dependencies
+
+- hive-exec-2.3.4.jar // contains hive-metastore-2.3.4
+
+
+// Hadoop dependencies
+- flink-shaded-hadoop-2-uber-2.7.5-1.8.0.jar
+- flink-hadoop-compatibility-{{site.version}}.jar
+
+```
+
+For Hive 1.2.1, users need:
+
+```
+// Hive dependencies
+
+- hive-metastore-1.2.1.jar
+- hive-exec-1.2.1.jar
+
+
+// Hadoop dependencies
+- flink-shaded-hadoop-2-uber-2.6.5-1.8.0.jar
+- flink-hadoop-compatibility-{{site.version}}.jar
+
+```
+
+If you don't have Hive dependencies at hand, they can be found at [mvnrepostory.com](https://mvnrepository.com):
+
+- [hive-exec](https://mvnrepository.com/artifact/org.apache.hive/hive-exec)
+- [hive-metastore](https://mvnrepository.com/artifact/org.apache.hive/hive-metastore)
+
+Note that users need to make sure the compatibility between their Hive versions and Hadoop versions. Otherwise, there may be potential problem, for example, Hive 2.3.4 is compiled against Hadoop 2.7.2, you may run into problems when using Hive 2.3.4 with Hadoop 2.4.
+
+
+## Data Type Mapping
+
+For both Flink and Hive tables, `HiveCatalog` stores table schemas by mapping them to Hive table schemas with Hive data types. Types are dynamically mapped back on read.
+
+Currently `HiveCatalog` supports most Flink data types with the following mapping:
+
+|  Flink Data Type  |  Hive Data Type  |
+|---|---|
+| CHAR(p)       |  char(p)* |
+| VARCHAR(p)    |  varchar(p)** |
+| STRING        |  string |
+| BOOLEAN       |  boolean |
+| BYTE          |  tinyint |
+| SHORT         |  smallint |
+| INT           |  int |
+| BIGINT        |  long |
+| FLOAT         |  float |
+| DOUBLE        |  double |
+| DECIMAL(p, s) |  decimal(p, s) |
+| DATE          |  date |
+| TIMESTAMP_WITHOUT_TIME_ZONE |  Timestamp |
+| TIMESTAMP_WITH_TIME_ZONE |  N/A |
+| TIMESTAMP_WITH_LOCAL_TIME_ZONE |  N/A |
+| INTERVAL |  N/A |
+| BINARY        |  binary |
+| VARBINARY(p)  |  binary |
+| ARRAY\<E>     |  list\<E> |
+| MAP<K, V>     |  map<K, V> |
+| ROW           |  struct |
+| MULTISET      |  N/A |
+
+
+Note that we only covers most commonly used data types for now.
+
+The following limitations in Hive's data types impact the mapping between Flink and Hive:
+
+\* maximum length is 255
+
+\** maximum length is 65535
+
+## Hive Compatibility
+
+For Hive compatibility and versions, see [Hive Compatibility]({{ site.baseurl }}/dev/batch/hive_compatibility.html)
+
+
+Catalog Registration
+--------------------
+
+## Register Catalog in Table API
+
+To register a catalog in Table API, users can create a catalog instance and register it through `TableEnvironment.registerCatalog(name, catalog)`.
+
+## Register Catalog in SQL CLI
+
+To use pre-defined catalogs (`GenericInMemoryCatalog` and `HiveCatalog`) in SQL CLI, please refer to [SQL Clinet]({{ site.baseurl }}/dev/table/sqlClient.html)
```

Comment:
```suggestion
To use pre-defined catalogs (`GenericInMemoryCatalog` and `HiveCatalog`) in the SQL CLI, please refer to its [documentation]({{ site.baseurl }}/dev/table/sqlClient.html).
```

Reply:
using `[<doc_name>]` seems to be a common practice in Flink doc.